src/main/java/com/faforever/client/i18n/I18nImpl.java

```diff
@@ -12,19 +15,49 @@
 
   @Inject
   Locale locale;
+  @Inject
+  ArrayList<String> languageInfo;
+  @Inject
+  PreferencesService preferencesService;
+
+  Locale userSpecificLocale;
+
+  @PostConstruct
+  public void init()
+  {
+    if(!preferencesService.getPreferences().getLang().getLanguage().equals("auto")) {
+      userSpecificLocale = new Locale(preferencesService.getPreferences().getLang().getLanguage(), getCountryFromLanguage(preferencesService.getPreferences().getLang().getLanguage()));
+    }else {
+      userSpecificLocale=locale;
+    }
+  }
+  @Override
+  public Locale getUserSpecificLocale()
+  {
+    return userSpecificLocale;
+  }
+
+  private String getCountryFromLanguage(String language) {
+    for (int i=0; i!=languageInfo.size();i++)
+    {
+      if(languageInfo.get(i).equals(language))return languageInfo.get(i+1);
+
+    }
+    return"UK";
```

Comment:
I did make it throw an exception and logged an error, but how do i get the default country from a language? Did not find that with in CountryCode....